builds/release/release-core-images.yaml

```diff
@@ -0,0 +1,540 @@
+            def filter: [
+              $an[] | . as $n | $aa[] | "mcr.microsoft.com/azureiotedge-\($n):\($v)-linux-\(.)"
+            ];
+            [ .[] | select(.product == $p and .version == $v) | .images[] ] | contains(filter)
+          ' $(Pipeline.Workspace)/detect-image-updates/image-updates/updates.json
+        )
+
+        echo "Continue with release? $continue"
+        echo "##vso[task.setvariable variable=continue;isOutput=true]$continue"
+      displayName: Filter image list for core images
+      name: filter
+
+    - task: AzureKeyVault@1
+      displayName: Get secrets
+      condition: and(succeeded(), eq(variables['filter.continue'], 'true'))
+      inputs:
+        azureSubscription: $(az.subscription)
+        keyVaultName: $(kv.name)
+        secretsFilter: TestGitHubAccessToken
+
+    - script: |
+        #!/bin/bash
+        set -euo pipefail
+
+        . scripts/linux/release-tools.sh
+
+        GIT_EMAIL='$(service-account.email)' \
+        BRANCH="${BUILD_SOURCEBRANCH#refs/heads/}" \
+        make_project_release_commit_for_core_image_refresh
+        get_project_release_info
+
+        changelog=$(echo "$OUTPUTS" | jq -rc '.changelog')
+        next=$(echo "$OUTPUTS" | jq -rc '.version')
+        prev=$(echo "$OUTPUTS" | jq -rc '.previous_version')
+        diag_version=$(echo "$OUTPUTS" | jq -rc '.diagnostics_version')
+        tags=$(echo "$OUTPUTS" | jq -rc '.tags')
+
+        echo "##vso[task.setvariable variable=changelog;isOutput=true]$changelog"
+        echo "##vso[task.setvariable variable=version;isOutput=true]$next"
+        echo "##vso[task.setvariable variable=previous_version;isOutput=true]$prev"
+        echo "##vso[task.setvariable variable=diagnostics_version;isOutput=true]$diag_version"
+        echo "##vso[task.setvariable variable=tags;isOutput=true]$tags"
+      displayName: Create a release commit
+      name: commit
+      condition: and(succeeded(), eq(variables['filter.continue'], 'true'))
+      env:
+        GITHUB_TOKEN: "$(TestGitHubAccessToken)"
+
+  - job: ManualTrigger
+    displayName: Manually trigger pipeline to update images
+    condition: eq(variables['Build.Reason'], 'Manual')
+
+    steps:
+    - checkout: self
+      fetchDepth: 0
+
+    - script: |
+        # Create a filter.continue task variable to match the PipelineTrigger job, but hard-code it
+        # to true so the pipeline will always run subsequent stages in the manual case.
+        echo "##vso[task.setvariable variable=continue;isOutput=true]true"
+      name: filter
+
+    - script: |
+        #!/bin/bash
+        set -euo pipefail
+        
+        # this script assumes the following steps have already been taken:
+        # - a release commit has been pushed to the project repo, and is reachable from this run's
+        #   source commit
+        # - the release commit was tagged with the new version
+        # - the project repo's CHANGELOG.md has been updated with information about the new release
```

Comment:
What happens if one of these isn't done properly? Anything bad?

Reply:
Let's say 1.4.2 is the current release, and you want to release 1.4.3.

- If you didn't make a release commit, then the pipeline will detect 1.4.2 as the "new" release and basically re-release it. This technically isn't a great outcome--it means customers could end up with a mix of old and new 1.4.2 images across their devices. They might think all their devices are running the same bits, but technically they're not. The only difference is the base image, so it likely wouldn't be a problem, but theoretically it could lead to confusion if some devices are misbehaving, and others aren't.
- If you made a release commit but didn't tag it, then it's the same as the previous case. We would re-release the 1.4.2.
- If you made a release commit but didn't update the changelog, then we would post the previous release's changelog in the new GitHub release page. Not a big deal. You can always go back and edit the GitHub release page by hand.

Reply:
To mitigate the 1st two bullets from my previous comment, we basically want to add something like the "Are you sure?" dialog box that pops up when you delete a file. 🙂 I see two ways to do this:
1. Add back the "version" pipeline parameter and confirm that it matches the most recent tag in the repo. 
2. Add back an approval step.

Both of these would only happen in the manual case. I think I prefer the 1st option because it's more explicit. If we add an approval step, it would be important for the approver to look at the pipeline output and confirm that the proposed release version matches expectations, but people will probably just press the button without thinking.

Reply:
I suppose we could also just refuse to continue a manual pipeline if the most recent tag in the repo matches images that were already published to MCR.

Reply:
> I suppose we could also just fail the manual pipeline if the most recent tag in the repo matches images that were already published to MCR.

That sounds good to me

Reply:
Added a check to the `ManualTrigger` job that will abort the pipeline if the most recent tag in the repo matches a version that has already been released to MCR.